File: tests/fixtures/php_project/src/FileStorage.php
<?php

declare(strict_types=1);

namespace LspcmdFixture;

/**
 * Stores users in files (stub implementation).
 */
class FileStorage implements Storage
{
    /**
     * Creates a new FileStorage instance.
     *
     * @param string $basePath The base path for file storage
     */
    public function __construct(
        private string $basePath
    ) {}

    /**
     * Gets the base path.
     *
     * @return string The base path
     */
    public function getBasePath(): string
    {
        return $this->basePath;
    }

    /**
     * Saves a user to a file.
     *
     * @param User $user The user to save
     */
    public function save(User $user): void
    {
        // Stub implementation
    }

    /**
     * Loads a user from a file.
     *
     * @param string $email The email to search for
     * @return User|null The user if found
     */
    public function load(string $email): ?User
    {
        // Stub implementation
        return null;
    }

    /**
     * Deletes a user file.
     *
     * @param string $email The email of the user to delete
     * @return bool True if the user was deleted
     */
    public function delete(string $email): bool
    {
        // Stub implementation
        return false;
    }

    /**
     * Lists all users from files.
     *
     * @return User[] All users
     */
    public function list(): array
    {
        // Stub implementation
        return [];
    }
}

File: test/fixtures/java_project/src/main/java/com/example/FileStorage.java
package com.example;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * File-based storage implementation.
 * Stores users in files (stub implementation).
 */
public class FileStorage extends AbstractStorage {
    private String basePath;
    private Map<String, User> cache = new HashMap<>();

    /**
     * Creates a new file storage with the given base path.
     *
     * @param basePath The base directory for storing files
     */
    public FileStorage(String basePath) {
        super("file");
        this.basePath = basePath;
    }

    /**
     * Gets the base path for file storage.
     *
     * @return The base path
     */
    public String getBasePath() {
        return basePath;
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public void save(User user) {
        // Stub: just cache in memory
        cache.put(user.getEmail(), user);
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public User load(String email) {
        return cache.get(email);
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public boolean delete(String email) {
        return cache.remove(email) != null;
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public List<User> list() {
        return new ArrayList<>(cache.values());
    }
}
